lib/puppet/pops/loader/loader.rb

```diff
@@ -22,10 +22,16 @@
 module Puppet::Pops
 module Loader
 class Loader
+  attr_reader :loader_name
 
   # Describes the kinds of things that loaders can load
   LOADABLE_KINDS = [:func_4x, :func_4xpp, :type_pp, :resource_type_pp].freeze
 
+  # @param [String] name the name of the loader. Must be unique among all loaders maintained by a {Loader} instance
+  def initialize(loader_name)
+    @loader_name = loader_name
```

Comment:
Name should probably be frozen since it is a String (bad things will otherwise happen if the name is modified externally).